_posts/2020-11-08-introduction-to-functions-in-R.md

```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+title:  "Introduction to programming functions in R!"
+date:   2020-11-08 12:54:04 -0700
+categories: R topics
+---
+In this post you will learn the basics of functions in R.  This will include:
+what a function is, how to set up a function, and how to call for a function
+from another R script.
+
+# What is a function?
+In programming, a function is a subroutine that performs some calculation with
+given variables.  These variables are given by a program and a result is
+returned.
+
+You can think of this type of function the same way you would think about a
+mathematical one.  
+The function is a shorthand way to make different variables go through the
+same process without needing to rewrite the formula every time.  
+
+For instance: if you have the function y=x^2, and are given a list of x
+variables (1:5).  You are doing the same function 5 times with different
+variables giving different outputs.  Here the function would be running in
+a loop of y=1^2, y=2^2, y=3^2, y=4^2, and y=5^2.  
+
+The difference between this mathematical function and a programming function
+that you make is that the programming function is located on its own R script
+and can be called for from any other script contained in the same directory.   
+
+There are also functions built into R's base and the packages that you can
+install, however it is necessary to make your own functions at times to do
+specific tasks.
+
+# How to set up a function
+The basic outline for making a function in R is
+```ruby
+ `func_name <- function (argument){  
+    statement`  
+   `}`
+```
+The function name can be what ever you want.  If you
+want a function that reads in a csv file containing data on different states
+and then subset that data to look at a specific state, you might name it
+"look_at_state_data".  So now we have `func_name = look_at_state_data`.
+
+Next we need to assign a function to that name with variables (the argument
+  ).  You use the `<-` symbol to
+tell R that what ever follows the symbol is named what immediately proceeds it.
+To keep functions more dynamic, the variables should be named based on the
+type of input you will use.    
+For this example one of the variables  will be  
+`(input_file_name)` and
+`(subsetted_state)`.  
+
+We now have the function:
+```ruby
+  `look_at_state_data <- function(input_file_name,  
+                                  subsetted_state){
+
+                                      }`
+```                                      
+
+
+Next you need your statement.  Here is where you read in your csv file.  It is
+ also a good idea to give the file a new variable name.  Here you might call
+ this variable "all_state_data" as this file has all of the states in it.  
+The function now looks like:
+```ruby
+  `look_at_state_data <- function(input_file_name,  
+                                  subsetted_state){
+    all_state_data <- read.csv(input_file_name)
+    }`
+```                           
```

Comment:
It may be helpful to add a photo of an example csv here with some data so that people can have a visual of what you are talking about in each step.